File: Modules/ServerTypes/PerformanceAppraisal/CompanyProfileForm.ts
import { ImageUploadEditor, StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CompanyProfileForm {
    Picture: ImageUploadEditor;
    Title: StringEditor;
    Address: StringEditor;
    Tel: StringEditor;
    Website: StringEditor;
}

export class CompanyProfileForm extends PrefixedContext {
    static formKey = 'PerformanceAppraisal.CompanyProfile';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CompanyProfileForm.init)  {
            CompanyProfileForm.init = true;

            var w0 = ImageUploadEditor;
            var w1 = StringEditor;

            initFormType(CompanyProfileForm, [
                'Picture', w0,
                'Title', w1,
                'Address', w1,
                'Tel', w1,
                'Website', w1
            ]);
        }
    }
}

File: Modules/HumanResource/PerformanceAppraisal/CompanyProfile/CompanyProfileForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HRMSoftware.PerformanceAppraisal.Forms;

[FormScript("PerformanceAppraisal.CompanyProfile")]
[BasedOnRow(typeof(CompanyProfileRow), CheckNames = true)]
public class CompanyProfileForm
{
    [DisplayName("Logo")]
    public string Picture { get; set; }
    public string Title { get; set; }
    public string Address { get; set; }
    public string Tel { get; set; }
    public string Website { get; set; }
    
}